neuralpp/symbolic/expression.py

```diff
@@ -0,0 +1,151 @@
+from typing import Any, List
+from abc import ABC, abstractmethod
+
+
+class Expression(ABC):
+    @abstractmethod
+    # 'Expression': forward reference for recursive type reference https://peps.python.org/pep-0484/#forward-references
+    def subexpression(self) -> List['Expression']:
```

Comment:
Looks like `subexpression` could be a property instead of a method, what do you think? ;)

Reply:
I think we can do that, since `subexpression` is not recursive and thus not computation-heavy. It's not specified as a call instead of a property in the project plan. @rodrigodesalvobraz do you think we should change that?

Reply:
Yes, I think that's a good idea!